### PR TITLE
Add rm_acquire_image to prepare for ImageMagick 7 

### PR DESCRIPTION
**Pull request by dlemstra**
This also fixes some possible null reference issues and memory leaks.

### CODE REVIEW

File: ext/RMagick/rmutil.c [hunk 1]
```diff
@@ -688,7 +703,12 @@ rm_pixelpacket_to_color_name_info(Info *info, PixelPacket *color)
 
     my_info = info ? info : CloneImageInfo(NULL);
```
Comment (Watson1978) [inline ext/RMagick/rmutil.c]: (This is no relationship to PR)
I think we should raise an exception or assert if NULL was passed via `info` to notify coding miss.

All `rm_pixelpacket_to_color_name_info()` calling are paired with `Data_Get_Struct()`.
If `Data_Get_Struct()` returns `NULL` as `info` with `self` object, I think Ruby runtime has been broken...
https://github.com/rmagick/rmagick/blob/a828b4e36466ec87544211b5eee055270012c74e/ext/RMagick/rminfo.c#L491-L492

File: ext/RMagick/rmimage.c [hunk 2]
```diff
@@ -3923,7 +3923,7 @@ Image_constitute(VALUE class, VALUE width_arg, VALUE height_arg
     exception = AcquireExceptionInfo();
 
     // This is based on ConstituteImage in IM 5.5.7
-    image = AcquireImage(NULL);
+    image = rm_acquire_image((ImageInfo *) NULL);
     if (!image)
     {
         xfree(pixels.v);
```
Comment (Watson1978) [inline ext/RMagick/rmimage.c]: In additional, `(void) DestroyExceptionInfo(exception);` is required at here to avoid memory leak.

Comment (dlemstra) [reply ext/RMagick/rmimage.c]: This was removed in another PR. Will rebase this.

Comment (Watson1978) [reply ext/RMagick/rmimage.c]: Ah, i see.

**Pull request merged**